lib/xhr/XHRShim.js

```diff
@@ -0,0 +1,145 @@
+/*jslint vars: true, plusplus: true, devel: true, nomen: true, indent: 4, forin: true, maxerr: 50, regexp: true */
+/* global DOMParser */
+(function(global) {
+    "use strict";
+
+    // This script requires access to the transport to send
+    // file requests to the editor as commands
+    var transport = global._Brackets_LiveDev_Transport;
+
+    var XMLHttpRequest = global.XMLHttpRequest;
+    if(!XMLHttpRequest) {
+        console.error("[Brackets LiveDev] XMLHttpRequest not supported in preview");
+        return;
+    }
+
+    function sendMessage(msg) {
+        if(!transport) {
+            console.error("[Brackets LiveDev] No transport set");
+            return;
+        }
+
+        transport.send(JSON.stringify(msg));
+    }
+
+    function XMLHttpRequestLiveDev() {
+        var xhr = new XMLHttpRequest();
+        var requestUrl;
+        var abortCalled = false;
+
+        var open = xhr.open;
+        // The async parameter for XHRs to the local file system
+        // is ignored as we cannot support synchronous requests
+        // and async is implicit
+        xhr.open = function(method, url, async, user, password) {
+            if(!(/\:?\/\//.test(url) || /\s*data\:/.test(url))) {
```

Comment:
I don't think we should pull in https://github.com/humphd/brackets/blob/bramble/src/filesystem/impls/filer/lib/content.js#L104 because I would then need to inject the new script into the preview in a new `<script>` tag. But for only one use here, its probably not worth it. But I have updated the regex to match what is in `content.js`